pkg/zfs/zfs.go

```diff
@@ -1035,3 +1008,87 @@ func (z *zfs) Diff(filesystemID, snapshot, snapshotOrFilesystem string) ([]types
 
 	return sortedResult, nil
 }
+
+func (z *zfs) clearMounts(filesystem string) error {
+
+	f, err := os.Open("/proc/self/mountinfo")
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	r := bufio.NewReader(f)
+	mountPrefix := os.Getenv("MOUNT_PREFIX")
+
+	mountpoints, err := filterMountpoints(mountPrefix, filesystem, r)
+	if err != nil {
+		return fmt.Errorf("failed to get filesystem %s mountpoints, error: %s", filesystem, err)
+	}
+
+	for _, m := range mountpoints {
+
+		ns, err := namespaceForMount(m)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"error":         err,
+				"mountpoint":    m,
+				"filesystem_id": filesystem,
+			}).Error("clearMounts: failed to get namespace for mount")
+			continue
+		}
+
+		log.WithFields(log.Fields{
+			"filesystem": filesystem,
+			"mountpoint": m,
+			"namespace":  ns,
+		}).Info("clearMounts: unmounting..")
+
+		err = unmount(ns, m)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"error":      err,
+				"filesystem": filesystem,
+				"mountpoint": m,
+				"namespace":  ns,
+			}).Error("failed to clear mountpoint")
+			return fmt.Errorf("failed to clear filesystem %s mountpoint %s, error: %s", filesystem, m, err)
+		}
+	}
+
+	return nil
+}
+
+func unmount(ns, mountpoint string) error {
```

Comment:
did we not have an unmount bit of code elsewhere?

Reply:
this one unmounts from different namespaces :|